bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/shared/simple/SimpleRule.java

```diff
@@ -12,18 +12,306 @@
+        this.templateUID = templateUID;
+    }
+
+    @Override
+    @Nullable
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * This method is used to specify the {@link Rule}'s human-readable name.
+     *
+     * @param ruleName the {@link Rule}'s human-readable name, or {@code null}.
+     */
+    public void setName(@Nullable String ruleName) {
+        name = ruleName;
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return tags;
+    }
+
+    /**
+     * This method is used to specify the {@link Rule}'s assigned tags.
+     *
+     * @param ruleTags the {@link Rule}'s assigned tags.
+     */
+    @SuppressWarnings("null")
+    public void setTags(Set<String> ruleTags) {
+        tags = ruleTags != null ? ruleTags : new HashSet<>();
+    }
+
+    @Override
+    @Nullable
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * This method is used to specify human-readable description of the purpose and consequences of the
+     * {@link Rule}'s execution.
+     *
+     * @param ruleDescription the {@link Rule}'s human-readable description, or {@code null}.
+     */
+    public void setDescription(@Nullable String ruleDescription) {
+        description = ruleDescription;
+    }
+
+    @Override
+    public Visibility getVisibility() {
+        return visibility;
+    }
+
+    /**
+     * This method is used to specify the {@link Rule}'s {@link Visibility}.
+     *
+     * @param visibility the {@link Rule}'s {@link Visibility} value.
+     */
+    @SuppressWarnings("null")
+    public void setVisibility(Visibility visibility) {
+        this.visibility = visibility == null ? Visibility.VISIBLE : visibility;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.eclipse.smarthome.automation.Rule#getConfiguration()
+     */
+    @Override
+    public Configuration getConfiguration() {
+        return configuration;
+    }
+
+    /**
+     * This method is used to specify the {@link Rule}'s {@link Configuration}.
+     *
+     * @param ruleConfiguration the new configuration values.
+     */
+    @SuppressWarnings("null")
+    public void setConfiguration(Configuration ruleConfiguration) {
+        this.configuration = ruleConfiguration == null ? new Configuration() : ruleConfiguration;
+    }
+
+    @Override
+    public List<ConfigDescriptionParameter> getConfigurationDescriptions() {
+        return configDescriptions;
+    }
+
+    /**
+     * This method is used to describe with {@link ConfigDescriptionParameter}s
+     * the meta info for configuration properties of the {@link Rule}.
+     */
+    @SuppressWarnings("null")
+    public void setConfigurationDescriptions(List<ConfigDescriptionParameter> configDescriptions) {
+        this.configDescriptions = configDescriptions == null ? new ArrayList<>() : configDescriptions;
+    }
+
+    @Override
+    public List<Condition> getConditions() {
+        return conditions;
```

Comment:
We should return a copy, not the internal collection that could be modified by the caller.